### ISSUE TITLE
Running multiple cases through xdist gets stuck around 95% of the time

### ISSUE THREAD

**Issue by gova-i267** [pytest-dev/pytest-xdist#1005]
Hi, Ask for help~

2024-01-16 17:58:26.653 [gw6] [ 85%] FAILED test/xxxxxx/case
2024-01-16 17:58:27.014 test/xxxxxx/case
2024-01-16 17:59:35.797 [gw5] [ 86%] PASSED test/xxxxxx/case
2024-01-16 17:59:36.126 test/xxxxxx/case
2024-01-16 18:00:28.257 [gw0] [ 88%] FAILED test/xxxxxx/case 
2024-01-16 18:00:28.268 test/xxxxxx/case
2024-01-16 18:02:36.356 [gw1] [ 90%] FAILED test/xxxxxx/case
2024-01-16 18:02:45.082 [gw6] [ 91%] FAILED test/xxxxxx/case
2024-01-16 18:04:27.782 [gw0] [ 93%] PASSED test/xxxxxx/case
2024-01-16 18:04:27.795 test/xxxxxx/case
2024-01-16 18:08:26.971 [gw0] [ 95%] PASSED test/xxxxxx/case
2024-01-16 18:08:26.987 test/xxxxxx/case
2024-01-16 18:08:26.989 [gw0] [ 96%] SKIPPED test/xxxxxx/case
2024-01-16 18:08:26.991 test/xxxxxx/case

What information do I need to provide?

pytest-xdist==3.5.0

**Comment by gova-i267** [comment]
execnet==1.7.1

**Comment by james-certn** [comment]
What do you mean, 'gets stuck'? the pytest process/commandline just hangs, and never completes?

**Comment by gova-i267** [comment]
> What do you mean, 'gets stuck'? the pytest process/commandline just hangs, and never completes? What if you downgrade to 3.4.0 of pytest-xdist?

Yes, it hangs, and never completes. The probability of this problem is very high.

**Comment by james-certn** [comment]
Again: What if you downgrade to 3.4.0 of pytest-xdist? With what ptest-xdist version does the problem not surface/manifest?
